baselines/fedper/README.md

```diff
@@ -0,0 +1,158 @@
+
+**Dataset:** CIFAR10, FLICKR-AES. CIFAR10 will be partitioned based on number of classes for data that each client shall recieve e.g. 4 allocated classes could be [1, 3, 5, 9]. FLICKR-AES is an unbalanced dataset, so there we only apply random sampling. 
+
+**Training Hyperparameters:** The hyperparameters can be found in conf/base.yaml file which is the configuration file for the main script. 
+
+| Description | Default Value |
+| ----------- | ----- |
+| num_clients | 10 |
+| clients per round | 10 |
+| number of rounds | 50 |
+| client resources | {'num_cpus': 4, 'num_gpus': 1 }|
+| learning_rate | 0.01 |
+| batch_size | 128 |
+| optimizer | SGD |
+| algorithm | fedavg|
+
+**Stateful Clients:**
+In this Baseline (FedPer), we must store the state of the local client head while aggregation of body parameters happen at the server. Flower is currently making this possible but for the time being, we reside to storing client _head_ state in a folder called client_states. We store the values after each fit and evaluate function carried out on each client, and call for the state before executing these funcitons. Moreover, the state of a unique client is accessed using the client ID. 
+
+> NOTE: This is a work-around so that the local head parameters are not reset before each fit and evaluate. Nevertheless, it can come to change with future releases. 
+
+
+## Environment Setup
+
+To construct the Python environment follow these steps:
+
+```bash
+# Set Python 3.10
+pyenv local 3.10.6
+# Tell poetry to use python 3.10
+poetry env use 3.10.6
+
+# install the base Poetry environment
+poetry install
+
+# activate the environment
+poetry shell
+```
+## Running the Experiments
+```bash
+python -m fedper.main # this will run using the default settings in the `conf/base.yaml` 
+
+# When running models for flickr dataset, it is important to keep batch size at 4 or lower since some clients (for reproducing experiment) will have very few examples of one class
+```
+
+To reproduce figures:
+```bash
+# make fedper/run_scripts.sh executable
+chmod u+x fedper/run_scripts.sh
+# uncomment lines in script that you want to run, then  
+bash fedper/run_scripts.sh
+
+# this config can also be overriden from the CLI
+```
+
+What you need to change in configuration files: 
+```bash
+algorithm: fedavg, fedper # these are currently supported
+server_device: 'cuda:0', 'cpu'
+dataset.name: 'cifar10', 'flickr'
+num_classes: 10, 100, 5 # respectively 
+dataset.num_classes: 4, 8, 10 #for non-iid split assigning n num_classes to each client (these numbers for CIFAR10 experiments)
+model.name: mobile, resnet
+```
+
+To run multiple runs, one can also reside to `HYDRA`'s multirun option. 
+```bash
+# for CIFAR10
```

Comment:
The commands should use `model_name` instead of `model.name`

Reply:
Also, probably the best is to add `--config-name cifar10` to the commands so they load the correct config instead of `base.yaml`.